config/cmd/configure.go

```diff
@@ -89,6 +90,8 @@ container images that are not used as init containers.`)
 
 	flag.BoolVar(&validate, "validate", true, "validate the output configuration for correctness")
 
+	flag.StringVar(&data.KillAfter, "kill-after", "", "time allowed for pod to response")
```

Comment:
`KillAfter` is a duration in the format described by the time package. Maybe we should parse it as a time.Duration with `flag.DurationVar(&data.KillAfter, ...)`. I understand that it must be passed as a string to the environment variable, but there is no validation. You can use a time.Duration and convert back to a string with https://pkg.go.dev/time#Duration.String.

Consider this scenario: the user passes a bad duration string. For example, "400a" instead of "400s". If we just use a string, we won't know this is problematic with the deployment configuration itself. We will only find out after we have deployed to the cluster and start running tests.

I agree `string -> time.Duration -> string` appears redundant, but the first conversion occurs under-the-hood. And, if it provides validation... I think that is an improvement. The other choice would be to manually invoke time.ParseDuration and error check. But, using the duration flag as intended feels less hacky to me.

Reply:
I see @paulosjca covered this in https://github.com/grpc/test-infra/pull/172/files#r645650206 as well. I am in agreement. Since Go allows Durations as a native type, I think we should utilize them. Interestingly, the `time.ParseDuration` function grammar is very similar to the `timeout --killafter` grammar if not identical.

Reply:
You are making a great point! I have learned that we should always make some effort if possible to validate input information as early as possible, I should not trust the input too much. I realized the validation is really important!